controllers/wizard/fileUpload/PKPFileUploadWizardHandler.inc.php

```diff
@@ -408,7 +522,12 @@ function confirmRevision($args, $request) {
 		// FIXME?: need assocType and assocId? Not sure if they would be used, so not adding now.
 		$reviewRound = $this->getReviewRound();
 		$confirmationForm = new SubmissionFilesUploadConfirmationForm(
```

Comment:
(I'd also like to come up with a better solution to function parameter lists that grow like this. We have some ideas for the DAOs and the service classes solve some problems using `$arg` arrays, but introduce others at the same time. I'm sure there are some clever PHP7 tools that are in adoption in other projects already.)

Reply:
Agreed. The `$args` arrays are good in that they are extensible, which makes it easy to work across applications and plugins. But they're terrible for defining expected variable names/types, and autocomplete in IDEs. I think a lot of this is improved with more careful code design, but I think that one approach is to use DependencyInjection. That's what Pimple is for (which we use under the hood for Service classes), but I've never really got my head around it.